### PR TITLE
Add support for api gateway + consul namespaces

### PR DESCRIPTION
**Pull request by andrewstucki**
This PR, coinciding with the [corresponding PR in the api gateway codebase](https://github.com/hashicorp/consul-api-gateway/pull/91) adds support for namespace mapping of deployed gateways.

More context around the usage envisioned is in that PR, but for the most part this copies the same configuration as used in the connect injection and catalog sync helm configuration.

Tested via `bats`:

```bash
➜  consul-k8s git:(apigateway-namespace-mirroring) bats charts/consul/test/unit/api-gateway-controller-deployment.bats |
 grep "enable namespace"
ok 5 apiGateway/Deployment: enable namespaces
ok 6 apiGateway/Deployment: enable namespace mirroring
ok 7 apiGateway/Deployment: enable namespace mirroring prefixes
```

### CODE REVIEW

**Review by nathancoleman — APPROVED** [review]

**Review by mikemorris — COMMENTED** [review]
LGTM from the API Gateway side, would appreciate a final review from @hc-github-team-consul-ecosystem before merging.

**Review by ndhanushkodi — APPROVED** [review]
Looks good to me!

File: charts/consul/values.yaml [hunk 1]
```diff
@@ -2552,6 +2552,29 @@ apiGateway:
       # @type: string
       service: null
 
+  # [Enterprise Only] These settings manage the api gateway's interaction with
```
Comment (ishustava) [inline charts/consul/values.yaml:2555]:
```suggestion
  # [Enterprise Only] These settings manage the API gateway's interaction with
```

**Review by ishustava — APPROVED** [review]
Looks great!

**Pull request merged**